compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4717,6 +4722,17 @@ private void checkTypesForRecords(BLangWaitForAllExpr waitExpr, AnalyzerData dat
         }
     }
 
+    private boolean checkBinaryBitwiseOperations(BLangExpression valueExpr) {
+        if (valueExpr.getKind() == NodeKind.BINARY_EXPR
+                && ((BLangBinaryExpr) valueExpr).opKind == OperatorKind.BITWISE_OR) {
+            return true;
+        }
+        if (valueExpr.getKind() == NodeKind.GROUP_EXPR) {
```

Comment:
Shall we bring this to the top so nested expression check will be done first

Reply:
Updated. 893363cdd2b81d5e911d9169b575eeef3c0ac6ce